### ISSUE TITLE
Behavior change in pylint 2.16.0 when flagging superfluous-parens.

### ISSUE THREAD

**Issue by raeganbarker** [PyCQA/pylint#8173]
### Bug description

With pylint 2.16.0, the following are now flagged as `superfluous-parens`.

```python
"""Simplest example."""
x = (1 + 2) # Now flagged as superfluous-parens.
```

```python
"""Pandas example."""
import pandas as pd

data = {'col1': [1, 2, 3]}
df = pd.DataFrame(data)
df['col1'] = (df['col1'] * 2) # Now flagged as superfluous-parens.
```

These statements were not flagged as `superfluous-parens` in pylint 2.15.10. 

In looking at the release notes for pylint 2.16.0, the only item I found which seemed directly related was targeting updating `superfluous-parens` to work with string concatenation scenarios:
- https://github.com/PyCQA/pylint/issues/4792
- https://github.com/PyCQA/pylint/pull/7752

As the above scenarios are not related to string concatenation, is this change in behavior between pylint 2.15.10 and pylint 2.16.0 to be expected?

Thanks in advance for any information you can provide!

### Command used

```shell
pylint superfluous_parens_change_simple.py
pylint superfluous_parens_change_pandas.py
```

### Pylint output

```shell
superfluous_parens_change_simple.py:2:0: C0325: Unnecessary parens after '=' keyword (superfluous-parens)
superfluous_parens_change_pandas.py:6:0: C0325: Unnecessary parens after '=' keyword (superfluous-parens)
```

### Expected behavior

No output.

### Pylint version

```shell
Before Behavior Change Observed:
pylint 2.15.10
astroid 2.13.5
Python 3.9.16 (main, Dec  7 2022, 10:02:13) 

When Behavior Change Observed:
pylint 2.16.0
astroid 2.14.1
Python 3.9.16 (main, Dec  7 2022, 10:02:13)
```

**Comment by DanielNoord** [comment]
```python
"""Simplest example."""
x = (1 + 2) # Now flagged as superfluous-parens.
```

This is superfluous right?
```console
>>> x = (1 + 2)
>>> x
3
>>> x = 1 + 2
>>> x
3
```

**Comment by raeganbarker** [comment]
> ```python
> """Simplest example."""
> x = (1 + 2) # Now flagged as superfluous-parens.
> ```
> 
> This is superfluous right?
> 
> ```
> >>> x = (1 + 2)
> >>> x
> 3
> >>> x = 1 + 2
> >>> x
> 3
> ```
@DanielNoord 
I agree that it's superfluous.

The bigger question is more along the lines of is this change in behavior intended, because the release notes do not appear to communicate this change. The PR directly related to `superfluous-parens` for the release is described as dealing with string concatenation scenarios, which the `x = (1 + 2)` scenario above does not fall under.

**Comment by DanielNoord** [comment]
I guess we missed a changelog entry for a PR here. I don't think this is such an issue as it makes a pre-existing message better rather than making a message worse.
